test/e2e/kubectl/kubectl.go

```diff
@@ -1320,46 +1319,47 @@ metadata:
-
-				ginkgo.By("exposing timestamps")
-				out = framework.RunKubectlOrDie("log", pod.Name, containerName, nsFlag, "--tail=1", "--timestamps")
-				l := lines(out)
-				framework.ExpectEqual(len(l), 1)
-				words := strings.Split(l[0], " ")
-				gomega.Expect(len(words)).To(gomega.BeNumerically(">", 1))
-				if _, err := time.Parse(time.RFC3339Nano, words[0]); err != nil {
-					if _, err := time.Parse(time.RFC3339, words[0]); err != nil {
-						e2elog.Failf("expected %q to be RFC3339 or RFC3339Nano", words[0])
-					}
+			ginkgo.By("checking for a matching strings")
+			_, err := framework.LookForStringInLog(ns, podName, "", "/api/v1/namespaces/kube-system", framework.PodStartTimeout)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("limiting log lines")
+			out := framework.RunKubectlOrDie("log", podName, "", nsFlag, "--tail=1")
+			gomega.Expect(len(out)).NotTo(gomega.BeZero())
+			framework.ExpectEqual(len(lines(out)), 1)
+
+			ginkgo.By("limiting log bytes")
+			out = framework.RunKubectlOrDie("log", podName, "", nsFlag, "--limit-bytes=1")
+			framework.ExpectEqual(len(lines(out)), 1)
+			framework.ExpectEqual(len(out), 1)
+
+			ginkgo.By("exposing timestamps")
+			out = framework.RunKubectlOrDie("log", podName, "", nsFlag, "--tail=1", "--timestamps")
+			l := lines(out)
+			framework.ExpectEqual(len(l), 1)
+			words := strings.Split(l[0], " ")
+			gomega.Expect(len(words)).To(gomega.BeNumerically(">", 1))
+			if _, err := time.Parse(time.RFC3339Nano, words[0]); err != nil {
+				if _, err := time.Parse(time.RFC3339, words[0]); err != nil {
+					e2elog.Failf("expected %q to be RFC3339 or RFC3339Nano", words[0])
 				}
+			}
 
-				ginkgo.By("restricting to a time range")
-				// Note: we must wait at least two seconds,
-				// because the granularity is only 1 second and
-				// it could end up rounding the wrong way.
-				time.Sleep(2500 * time.Millisecond) // ensure that startup logs on the node are seen as older than 1s
-				recentOut := framework.RunKubectlOrDie("log", pod.Name, containerName, nsFlag, "--since=1s")
-				recent := len(strings.Split(recentOut, "\n"))
-				olderOut := framework.RunKubectlOrDie("log", pod.Name, containerName, nsFlag, "--since=24h")
-				older := len(strings.Split(olderOut, "\n"))
-				gomega.Expect(recent).To(gomega.BeNumerically("<", older), "expected recent(%v) to be less than older(%v)\nrecent lines:\n%v\nolder lines:\n%v\n", recent, older, recentOut, olderOut)
-			})
+			ginkgo.By("restricting to a time range")
+			// Note: we must wait at least two seconds,
+			// because the granularity is only 1 second and
+			// it could end up rounding the wrong way.
+			time.Sleep(2500 * time.Millisecond) // ensure that startup logs on the node are seen as older than 1s
+			recentOut := framework.RunKubectlOrDie("log", podName, "", nsFlag, "--since=1s")
+			recent := len(strings.Split(recentOut, "\n"))
+			olderOut := framework.RunKubectlOrDie("log", podName, "", nsFlag, "--since=24h")
```

Comment:
Interesting. The redis image won't output any new log lines after "Ready to accept connections", so the`recentOut` will have always have zero lines. With the new image, we may see non-zero lines, which is probably a good thing.